src/daft-core/src/array/ops/list.rs

```diff
@@ -114,6 +140,20 @@ impl ListArray {
             )))
         }
     }
+
+    pub fn contains(&self, elements: &Series) -> DaftResult<BooleanArray> {
+        let elements_iterator: Box<dyn Iterator<Item = Box<dyn arrow2::array::Array>>> =
+            if elements.len() == 1 {
+                Box::new(std::iter::repeat(elements.to_arrow()))
+            } else {
+                assert!(elements.len() == self.len());
+                Box::new((0..self.len()).map(|i| elements.inner.to_arrow().sliced(i, 1)))
```

Comment:
I think you actually want `as_arrow`: https://github.com/Eventual-Inc/Daft/blob/main/src/daft-core/src/array/ops/as_arrow.rs#L21. to_arrow is for exporting through ffi

Reply:
Also once you call `as_arrow` you should be able to call `.into_iter()` on the concrete type `https://github.com/jorgecarleitao/arrow2/blob/main/src/array/fixed_size_list/iterator.rs`

Reply:
I can't do `as_arrow` because it is only defined over `DataArray`, and I can't get the `DataArray` for `elements` because I don't know the concrete type of `elements.inner` without doing a big `match` statement.

Is there a better workaround here?

Reply:
I guess I could add an `Series::as_arrow()`, but that feels like it would be pretty confusing vs `Series::to_arrow()`?

Reply:
I would probably go that route since that would be correct. you can do a with_match_physical after converting it to the physical array. and Since you know that self.child and elements are the same dtype, it should be just be 1 match.

Reply:
> I would probably go that route

Are you referring to my suggestion: " I can't get the DataArray for elements because I don't know the concrete type of elements.inner without doing daft_match... statement on the dtype."?

I just tried your suggestion, but am running into more problems

```rust
let elements = elements.as_physical()?;
with_match_physical_daft_types!(elements.data_type(), |$T| {
    let elements = elements.downcast::<$T>()?;
    elements.as_arrow() // <- Failing here because of ExtensionType, which seems to be a PhysicalType but I guess it doesn't map directly to an arrow2 concrete type array?
})
```

In general our codebase feels really hard to work with when we have more than one level of types and the nested types are unknown... Super gnarly.